### ISSUE TITLE
[Linux - Mesa 23.0.3] Shaders fail to compile

### ISSUE THREAD

**Issue by SpidFightFR** [IrisShaders/Iris#1967]
### What happened?

When loading shaders (SEUS PTGI E12 & PTGI HRR Test 2.1 GFME v1.12 were tested here), Iris don't seem to be able to compile shaders, and errors are displayed in the logs:

<details>

<summary>Errors</summary>

```
ac_rtld error(1): too much LDS (used = 35784, max = 32768)
LLVM failed to upload shader
EE ../mesa-23.0.3/src/gallium/drivers/radeonsi/si_state_shaders.cpp:2557 si_build_shader_variant - Failed to build shader variant (type=3)
ac_rtld error(1): too much LDS (used = 35784, max = 32768)
LLVM failed to upload shader
EE ../mesa-23.0.3/src/gallium/drivers/radeonsi/si_state_shaders.cpp:2557 si_build_shader_variant - Failed to build shader variant (type=3)
[20:26:52] [Render thread/INFO]: World Chunk Loading: 239/28
ac_rtld error(1): too much LDS (used = 35784, max = 32768)
LLVM failed to upload shader
EE ../mesa-23.0.3/src/gallium/drivers/radeonsi/si_state_shaders.cpp:2557 si_build_shader_variant - Failed to build shader variant (type=3)
```

</details>

And as a result, the game renders weirdly.

### Screenshots

[Screenshot_PTGI_E12](https://i.imgur.com/OqF7QUe.jpg) and [GFME_v1.12](https://i.imgur.com/j5iYZ7K.jpg)

### Log output

Logs available [here](https://mclo.gs/xlDjqIK).

### Minecraft Version

Minecraft 1.19.4

### Iris Version

Iris 1.6.1

### Operating System

Linux (Archlinux - kernel 6.2.12-zen1-1-zen - Mesa 23.0.3)

### What is your GPU?

AMD RX 7900 XTX

### Additional context

Tried both quilt and fabric on their latest stable releases, even tried with an older version of Iris (just in case) and it still happens. My best guess is that Mesa 23.0.3 broke a compat with shaders compilation.

**Comment by NinjaCheetah** [comment]
Having the same issue (similar specs, Iris 1.6.1/Mesa 23.0.3/Linux Kernel 6.2.13). Areas the sun can "access" (anywhere that connects to the sky without blocks in the way) is full bright and anywhere that's sealed off is completely black with no lighting being traced.

**Comment by Hunkarada** [comment]
Kernel 6.2-pf6, Mesa 22.3.7, Iris 1.5.2, works perfectly on AMD videocard (RX 580 2048sp). Need to check, where is a bug, in mesa or in Iris, maybe I'll update mesa soon and tell about a results.

**Comment by Hunkarada** [comment]
As I found, Mesa at 23 ver. add shader meshes, and optimized RDNA3 arch (AMD RX 7900 XTX is RDNA3). Updating mesa to 23 from 22 on Gentoo. Btw mesa fixed something (idk what) in minecraft... (according to opennet news)

**Comment by Hunkarada** [comment]
Updated mesa to 23.0.3, everything works perfectly, mesa fixed for minecraft this: `Minecraft: spec related compile errors`
Iris should be bugged, somewhere between 1.5.2 to 1.6.1

**Comment by Funtimes909** [comment]
You could always try the git version of mesa

**Comment by Hunkarada** [comment]
I've installed the same version as @SpidFightFR, so problem isn't in mesa (btw, I'm using gentoo, and MAYBE I compiled mesa without buggy code with -O3 optimization, so the best way to check where is a bug is find another arch user)

**Comment by Lolothepro** [comment]
probably related to this: https://gitlab.freedesktop.org/mesa/mesa/-/issues/9048

**Comment by SpidFightFR** [comment]
This issue has been solved with mesa 23.1.
